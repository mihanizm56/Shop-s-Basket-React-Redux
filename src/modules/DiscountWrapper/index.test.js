import React from "react"
import { shallow } from "enzyme"
import { DiscountWrapper } from "./index"

describe("DiscountWrapper test", () => {
  const userItemsTest = [
    {
      img: "./img/img1.png",
      description: "Платье-миди с расклешенной юбкой",
      code: "82039-11",
      size: "44",
      color: "синий",
      price: 2450,
      numberOfItems: 1,
      id: 1,
    },
    {
      img: "./img/img2.png",
      description: "Туфли женские украшенные кружевными вставками",
      code: "13524-01",
      size: "38",
      color: "черный",
      price: 2450,
      numberOfItems: 1,
      id: 2,
    },
    {
      img: "./img/img3.png",
      description: "Платье-миди",
      code: "75039",
      size: "44",
      color: "белый",
      price: 2450,
      numberOfItems: 1,
      id: 3,
    },
    {
      img: "./img/img4.png",
      description: "Платье-миди",
      code: "75039",
      size: "46",
      color: "белый",
      price: 2450,
      numberOfItems: 1,
      id: 4,
    },
  ]

  it("Renders correct without props", () => {
    const component = shallow(<DiscountWrapper />)

    expect(component).toMatchSnapshot()
  })
  it("Renders correct with props", () => {
    const component = shallow(
      <DiscountWrapper
        promocode="123"
        userItems={userItemsTest}
        discount={true}
        checkDiscount={() => "test_callback"}
        discountValue={2000}
      />
    )

    expect(component).toMatchSnapshot()
  })
})
