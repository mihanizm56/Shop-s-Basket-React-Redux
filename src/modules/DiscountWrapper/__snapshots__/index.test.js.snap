// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="promo-wrapper"
>
  <div
    className="promo-enter"
  >
    <p
      className="promo-enter__title"
    >
      Есть промокод?
    </p>
    <div
      className="input-block"
    >
      <input
        className="input-block__input"
        onChange={[Function]}
        placeholder="Введите промокод"
        type="text"
      />
      <button
        className="input-block__button"
        style={
          Object {
            "backgroundColor": "#FF6D00",
          }
        }
      >
        Применить
      </button>
    </div>
  </div>
  <div
    className="check-sum"
  >
    <div
      className="sum-display"
    >
      <div
        className="first-text-block"
      >
        <span
          className="first-text-block-text first-text-block__sum-text"
        >
          Сумма заказа:
        </span>
        <span
          className="first-text-block-text first-text-block__promo-text"
        >
          Промокод:
        </span>
        <span
          className="first-text-block-text first-text-block__result-text"
        >
          Всего:
        </span>
      </div>
      <div
        className="second-text-block"
      >
        <span
          className="second-text-block-text second-text-block__sum-value"
        >
          9800
           руб.
        </span>
        <span
          className="second-text-block-text second-text-block__promo-value"
        >
          2000 руб.
        </span>
        <span
          className="second-text-block-text second-text-block__result-value"
        >
          7800
           
          руб.
        </span>
      </div>
    </div>
    <button
      className="check-sum__button"
    >
      Оформить заказ
    </button>
  </div>
</div>
`;

exports[`DiscountWrapper test Renders correct with props 1`] = `
<div
  className="promo-wrapper"
>
  <div
    className="promo-enter"
  >
    <p
      className="promo-enter__title"
    >
      Есть промокод?
    </p>
    <div
      className="input-block"
    >
      <input
        className="input-block__input"
        onChange={[Function]}
        placeholder="Введите промокод"
        type="text"
      />
      <button
        className="input-block__button"
        style={
          Object {
            "backgroundColor": "#FF6D00",
          }
        }
      >
        Применить
      </button>
    </div>
  </div>
  <div
    className="check-sum"
  >
    <div
      className="sum-display"
    >
      <div
        className="first-text-block"
      >
        <span
          className="first-text-block-text first-text-block__sum-text"
        >
          Сумма заказа:
        </span>
        <span
          className="first-text-block-text first-text-block__promo-text"
        >
          Промокод:
        </span>
        <span
          className="first-text-block-text first-text-block__result-text"
        >
          Всего:
        </span>
      </div>
      <div
        className="second-text-block"
      >
        <span
          className="second-text-block-text second-text-block__sum-value"
        >
          9800
           руб.
        </span>
        <span
          className="second-text-block-text second-text-block__promo-value"
        >
          2000 руб.
        </span>
        <span
          className="second-text-block-text second-text-block__result-value"
        >
          7800
           
          руб.
        </span>
      </div>
    </div>
    <button
      className="check-sum__button"
    >
      Оформить заказ
    </button>
  </div>
</div>
`;

exports[`DiscountWrapper test Renders correct without props 1`] = `
<div
  className="promo-wrapper"
>
  <div
    className="promo-enter"
  >
    <p
      className="promo-enter__title"
    >
      Есть промокод?
    </p>
    <div
      className="input-block"
    >
      <input
        className="input-block__input"
        onChange={[Function]}
        placeholder="Введите промокод"
        type="text"
      />
      <button
        className="input-block__button"
        style={Object {}}
      >
        Применить
      </button>
    </div>
  </div>
  <div
    className="check-sum"
  >
    <div
      className="sum-display"
    >
      <div
        className="first-text-block"
      >
        <span
          className="first-text-block-text first-text-block__sum-text"
        >
          Сумма заказа:
        </span>
        <span
          className="first-text-block-text first-text-block__result-text"
        >
          Всего:
        </span>
      </div>
      <div
        className="second-text-block"
      >
        <span
          className="second-text-block-text second-text-block__sum-value"
        >
           руб.
        </span>
        <span
          className="second-text-block-text second-text-block__promo-value"
        />
        <span
          className="second-text-block-text second-text-block__result-value"
        >
           
          руб.
        </span>
      </div>
    </div>
    <button
      className="check-sum__button"
    >
      Оформить заказ
    </button>
  </div>
</div>
`;
